core/modules/replica/RequestMessenger.h

```diff
@@ -0,0 +1,114 @@
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_REQUEST_MESSENGER_H
+#define LSST_QSERV_REPLICA_REQUEST_MESSENGER_H
+
+/// RequestMessenger.h declares:
+///
+/// class RequestMessenger
+/// (see individual class documentation for more information)
+
+// System headers
+#include <memory>
+#include <string>
+
+// Third party headers
+#include <boost/asio.hpp>
+
+// Qserv headers
+#include "replica/Request.h"
+#include "replica/ServiceProvider.h"
+
+// Forward declarations
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+// Forward declarations
+class Messenger;
+
+/**
+  * Class RequestMessenger is a base class for a family of requests within
+  * the replication Controller server.
+  */
+class RequestMessenger
+    :   public Request  {
+
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<RequestMessenger> pointer;
+
+    // Default construction and copy semantics are prohibited
+
+    RequestMessenger() = delete;
+    RequestMessenger(RequestMessenger const&) = delete;
+    RequestMessenger& operator=(RequestMessenger const&) = delete;
+
+    /// Destructor
+    ~RequestMessenger() override = default;
+
+protected:
+
+    /**
+     * Construct the request with the pointer to the services provider.
+     *
+     * NOTE: options 'keepTracking' and 'allowDuplicate' have effect for
+     *       specific request only.
+     *
+     * @param serviceProvider - a provider of various services
+     * @param type            - its type name (used informally for debugging)
+     * @param worker          - the name of a worker
+     * @io_service            - BOOST ASIO service
+     * @priority              - may affect an execution order of the request by
+     *                          the worker service. Higher number means higher
+     *                          priority.
+     * @param keepTracking    - keep tracking the request before it finishes or fails
+     * @param allowDuplicate  - follow a previously made request if the current one duplicates it
+     * @param messenger       - an interface for communicating with workers
+     */
+    RequestMessenger(ServiceProvider::pointer const& serviceProvider,
+                     boost::asio::io_service& io_service,
+                     std::string const& type,
+                     std::string const& worker,
+                     int  priority,
+                     bool keepTracking,
+                     bool allowDuplicate,
+                     std::shared_ptr<Messenger> const& messenger);
+
+    /**
+     * Implement a method defined in the base class.
+     */
+    void finishImpl() override;
+    
+protected:
+
+    /// Worker messenging service
+    std::shared_ptr<Messenger> _messenger;
+};
+
+}}} // namespace lsst::qserv::replica
+
+#endif // LSST_QSERV_REPLICA_REQUEST_MESSENGER_H
```

Comment:
missing newline

Reply:
can you set your IDE to automatically add a newline at the end of the file? At least some support this.